pandas/core/groupby.py

```diff
@@ -468,6 +473,7 @@ def _set_selection_from_grouper(self):
 
             if len(groupers):
                 self._group_selection = ax.difference(Index(groupers)).tolist()
+                self._reset_cache('_selected_obj')
```

Comment:
hmm I think *maybe* move the ``_reset_cache`` higher up (e.g. outside the ``len(groupers)`` though not sure if this breaks anything (it just feels cleaner)